src/snowflake/snowpark/dataframe_analytics_functions.py

```diff
@@ -248,3 +371,144 @@ def cumulative_agg(
+        ...     ["2023-01-02", 101, 100],
+        ...     ["2023-01-03", 101, 300],
+        ...     ["2023-01-04", 102, 250],
+        ... ]
+        >>> df = session.create_dataframe(sample_data).to_df(
+        ...     "ORDERDATE", "PRODUCTKEY", "SALESAMOUNT"
+        ... )
+        >>> df = df.with_column("ORDERDATE", to_timestamp(df["ORDERDATE"]))
+        >>> def custom_formatter(input_col, agg, window):
+        ...     return f"{agg}_{input_col}_{window}"
+        >>> res = df.analytics.time_series_agg(
+        ...     time_col="ORDERDATE",
+        ...     group_by=["PRODUCTKEY"],
+        ...     aggs={"SALESAMOUNT": ["SUM", "MAX"]},
+        ...     windows=["1D", "-1D"],
+        ...     sliding_interval="12H",
+        ...     col_formatter=custom_formatter,
+        ... )
+        >>> res.show()
+        --------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+        |"PRODUCTKEY"  |"SLIDING_POINT"      |"SALESAMOUNT"  |"ORDERDATE"          |"SUM_SALESAMOUNT_1D"  |"MAX_SALESAMOUNT_1D"  |"SUM_SALESAMOUNT_-1D"  |"MAX_SALESAMOUNT_-1D"  |
+        --------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+        |101           |2023-01-01 00:00:00  |200            |2023-01-01 00:00:00  |300                   |200                   |200                    |200                    |
+        |101           |2023-01-02 00:00:00  |100            |2023-01-02 00:00:00  |400                   |300                   |300                    |200                    |
+        |101           |2023-01-03 00:00:00  |300            |2023-01-03 00:00:00  |300                   |300                   |400                    |300                    |
+        |102           |2023-01-04 00:00:00  |250            |2023-01-04 00:00:00  |250                   |250                   |250                    |250                    |
+        --------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+        <BLANKLINE>
+        """
+        self._validate_aggs_argument(aggs)
+        self._validate_string_list_argument(group_by, "group_by")
+        self._validate_formatter_argument(col_formatter)
+
+        if not windows:
+            raise ValueError("windows must not be empty")
+
+        if not sliding_interval:
+            raise ValueError("sliding_interval must not be empty")
+
+        if not time_col or not isinstance(time_col, str):
+            raise ValueError("time_col must be a string")
+
+        slide_duration, slide_unit = self._validate_and_extract_time_unit(
+            sliding_interval, "sliding_interval", allow_negative=False
+        )
+        sliding_point_col = "sliding_point"
+
```

Comment:
would this overwrite sliding_point if original df has this column? I would suggest using randomized name - see `random_table_name` in `utils.py`

Reply:
This is not a temporary column. I am not sure about generating a random name for an output column. This will make it hard to write queries based on this column. We could put a contact suffix/prefix to the name. That can bring down the probability of collision. Say "sp_gen_sliding_point". wdyt ?